pkg/provisioner/ironic/testserver/ironic.go

```diff
@@ -87,6 +90,71 @@ func (m *IronicMock) WithNode(node nodes.Node) *IronicMock {
 	return m
 }
 
+// WithWritableNode configures the server with a valid response for /v1/nodes
+func (m *IronicMock) WithWritableNode(node *nodes.Node) *IronicMock {
```

Comment:
Probably it would be better to split into different methods the GET/PATCH for /v1/nodes, as they belong to different api calls (and for the get we already have a working implementation)

Reply:
I just realized that I need a similar feature while testing the `Delete` method, let me workout a proposal in a separate PR. For the moment I think it's ok to proceed as it is.